proposals/0132-sequence-end-ops.md

```diff
@@ -16,724 +17,227 @@ these names so they follow consistent, predictable patterns.
-| startIndex ... (i: Index)                        | through: Index               | C.prefix(through:&nbsp;i) | -                                   | -                                                           |
-
-### Why does it matter?
-
-Many of these APIs are only occasionally necessary, so it's important 
-that they be easy to find when needed and easy to understand when 
-read. If you know that `prefix(10)` will get the first ten elements but 
-don't know what its inverse is, you will probably not guess that it's 
-`dropFirst(10)`. The confusing, conflicting names in these APIs are a 
-barrier to users adopting them where appropriate.
+  [onesided]: (0172-one-sided-ranges.md)
+  
+| Operation        | Operand                           | Current name                  |
+| ---------------- | --------------------------------- | ----------------------------- |
+| Get value        | First element                     | `first`                       |
+| "                | Last element                      | `last`                        |
+| "                | First *n* elements                | `prefix(3)`                   |
+| "                | Last *n* elements                 | `suffix(3)`                   |
+| "                | Leading elements matching closure | `prefix(while: isOdd)`        |
+| "                | Earliest element matching closure | `first(where: isOdd)`         |
+| Get index        | Earliest element equal to value   | `index(of: x)`                |
+| "                | Earliest element matching closure | `index(where: isOdd)`         |
+| Remove from copy | First element                     | `dropFirst()`                 |
+| "                | Last element                      | `dropLast()`                  |
+| "                | First *n* elements                | `dropFirst(3)`                |
+| "                | Last *n* elements                 | `dropLast(3)`                 |
+| "                | Leading elements matching closure | `drop(while:)`                |
+| Remove from self | First element                     | `removeFirst()`               |
+| "                | Last element                      | `removeLast()`                |
+| "                | First *n* elements                | `removeFirst(3)`              |
+| "                | Last *n* elements                 | `removeLast(3)`               |
+| …if present      | First element                     | `popFirst()`                  |
+| "                | Last element                      | `popLast()`                   |
+| Test equality    | First *n* elements                | `starts(with: other)`         |
+| …with function   | "                                 | `starts(with: other, by: ==)` |
+
+Put next to each other, we see a lot of inconsistent terminology:
+
+* Usually, "first *n* elements" is handled by overloading a `first` method, except 
```

Comment:
I don’t see any justification for that claim.  There’s only one `first` method in your list, `first(where:)`.

Reply:
I’m also speaking of `dropFirst(n)`, `removeFirst(n)`, etc. This could probably be written more clearly.